src/gamepad/gamepad-mappings-manager.vala

```diff
@@ -22,6 +22,13 @@ private class Games.GamepadMappingsManager : Object {
 		catch (Error e) {
 			warning ("GamepadMappingsManager: Can't find gamecontrollerdb.txt: %s", e.message);
 		}
+		try {
+			var dir = Application.get_config_dir ();
+			add_from_file (@"$dir/gamecontrollerdb.txt");
```

Comment:
It's probably better to check if the file exist here: from a semantic POV `add_from_file` sounds like it expects an existing file.